### DISCUSSION TITLE
disable changing the field Email on the account page

### DISCUSSION THREAD

**Discussion by ayoubAnbara** [keycloak/keycloak#9197] [Q&A]
I want to disable the email changes when users log into their client account area.

![client account area](https://aws1.discourse-cdn.com/free1/uploads/keycloak/optimized/1X/e36a3049ee46deae3941737b1840d20ee912f691_2_690x189.png)

PS: 
1- I don't want to do it by making the field email read-only in the HTML form in my customized theme.
2- [Link of another discussion in keycloak discourse group about the same question ](https://keycloak.discourse.group/t/hide-disable-email-changes-on-client-account-area/2241)

**Comment by cgeorgilakis** [comment]
This is something we also must have.
We plan to try to achieve it with [User profile](https://www.keycloak.org/docs/latest/server_admin/index.html#user-profile).
User profile is Technology Preview and is not fully supported. You must enable it with -Dkeycloak.profile=preview or -Dkeycloak.profile.feature.declarative_user_profile=enabled and set User Profile Enabled to true.
We see for documentations that email exists in attributes tab of User Profile.

When we test it, I will inform you. Does Keycloak team has any other information about it and the plan for making User Profile officially supporting? User profile has many useful features for User Attributes.

**Reply by cgeorgilakis** [reply]
I have tested it with User Profile in Keycloak 16.1.0 . Unfortunately, we can not configure permissions in main attributes (fe email) like in other attributes.

Keycloak team could you add the functionality for changing permissions in main attributes like in other attributes?
It is important for our team the ability of User to only change his first name - last name and not email.
